server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -672,4 +661,146 @@ boolean waitForBackgroundTermination(long timeout) throws InterruptedException
   {
     return backgroundManagerExitedLatch.await(timeout, TimeUnit.MILLISECONDS);
   }
+
+  public static class LookupsCommunicator
```

Comment:
Consider moving this class to top level, LookupCoordinatorManager is very big already.